packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -84,6 +113,25 @@ class FlutterPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
+        project.extensions.create("flutter", FlutterExtension)
+        project.afterEvaluate this.&addFlutterTask
+
+        allTargetPlatforms.each { currentTargetPlatformValue, abi ->
+            project.android {
+                packagingOptions {
+                    pickFirst "lib/${abi}/libflutter.so"
+
+                    // Disable warning by *-android-strip: File format not recognized
+                    doNotStrip "*/${abi}/lib_vm_snapshot_data.so"
+                    doNotStrip "*/${abi}/lib_vm_snapshot_instr.so"
+                    doNotStrip "*/${abi}/lib_isolate_snapshot_data.so"
+                    doNotStrip "*/${abi}/lib_isolate_snapshot_instr.so"
+                }
+            }
+        }
+
+        String targetPlatform = getTargetPlatform(project)
```

Comment:
nit: It's not obvious here that this can be null, and we're handling that case a few dozen lines below. I think it would be a little bit clearer if `getTargetPlatform` returned a default instead of null.